### ISSUE TITLE
[DevTools] Make sure named hooks Suspense cache is persisting between elements

### ISSUE THREAD

**Issue by bvaughn** [facebook/react#21822]
It currently looks like the `map.get(element)` returns `undefined` even for previously-loaded elements. Are we evicting from the cache prematurely? Why?

Our second level caching (inside of `parseHookNames`) is working but this is still resulting in a lot of unnecessary work.

**Comment by bvaughn** [comment]
Update: I am seeing what looks items getting evicted from the Suspense cache unexpectedly.

I added logging and verified that the elements are referentially equal between selections, but the cache is `returning` undefined – (the `WeakMap` has fewer items in it than it should). I changed from a `WeakMap` to a `Map` (even though the objects aren't being released so I wouldn't expect a difference) and the behavior is unchanged. I also added another `WeakMap` (not connected to the Suspense cache, just one managed by my module) and it retains the elements between selections, so it seems like the Suspense cache map is specifically doing something unexpected here.
